pkg/registry/controller/rest.go

```diff
@@ -47,6 +47,12 @@ func (rcStrategy) NamespaceScoped() bool {
 func (rcStrategy) PrepareForCreate(obj runtime.Object) {
 	controller := obj.(*api.ReplicationController)
 	controller.Status = api.ReplicationControllerStatus{}
+	if len(controller.Spec.Selector) == 0 {
+		controller.Spec.Selector = controller.Spec.Template.Labels
+	}
+	if len(controller.Labels) == 0 {
+		controller.Labels = controller.Spec.Template.Labels
+	}
 }
```

Comment:
GracePeriodSeconds is similar; it was made a pointer:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/api/v1beta3/types.go#L1234